pkg/controller/framework/controller.go

```diff
@@ -102,3 +103,117 @@ func (c *Controller) processLoop() {
 		}
 	}
 }
+
+// ResourceEventHandler can handle notifications for events that happen to a resource.
+// The events are informational only, so you can't return an error.
+// OnDelete will get the final state of the item if it is known, otherwise it will get
+// an object of type cache.DeletedFinalStateUnknown.
+type ResourceEventHandler interface {
+	OnAdd(obj interface{})
+	OnUpdate(obj interface{})
+	OnDelete(obj interface{})
+}
+
+// ResourceEventHandlerFuncs is an adaptor to let you easily specify as many
+// notification functions as you want while still implementing
+// ResourceEventHandler.
+type ResourceEventHandlerFuncs struct {
```

Comment:
I didn't understand the part about "as many notification functions as you want." It looks to me like you specify at most one Add, one Update, and one Delete function?